syntax/printer.go

```diff
@@ -1079,11 +1079,22 @@ func (p *Printer) command(cmd Command, redirs []*Redirect) (startRedirs int) {
 		p.ifClause(x, false)
 	case *Subshell:
 		p.WriteByte('(')
-		if len(x.Stmts) > 0 && startsWithLparen(x.Stmts[0]) {
+		stmts := x.Stmts
+		if len(stmts) > 0 && startsWithLparen(stmts[0]) {
 			p.wantSpace = spaceRequired
+			// we only want to keep the space between two nested subshells' open brackets
+			// if its in a single line to avoid ambiguity
+			if x.Lparen.Line() != stmts[0].Pos().Line() || len(stmts) > 1 && !p.singleLine {
+				p.wantSpace = spaceNotRequired
+
+				if p.minify {
+					p.newline(stmts[0].Pos())
```

Comment:
ah, so this was a bug? can we not just add it to the condition above, like singleLine?